homework/index.js

```diff
@@ -23,24 +23,147 @@
       const value = options[key];
       if (key === 'text') {
         elem.textContent = value;
+      } else if (key === 'html') {
+        elem.innerHTML = value;
       } else {
         elem.setAttribute(key, value);
       }
     });
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, repositories) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  const renderError = error => {
+    const wrapper = document.getElementById('wrapper');
+    wrapper.innerHTML = '';
+    createAndAppend('div', wrapper, {
+      text: error.message,
+      class: 'alert',
+    });
+  };
+
+  const addRepoInfoRow = (parent, label, content) => {
+    const row = createAndAppend('tr', parent);
+    createAndAppend('th', row, {
+      text: label,
+      class: 'table-header',
+    });
+    createAndAppend('td', row, {
+      html: content,
+    });
+  };
+
+  function renderRepoInfo(repository) {
+    const leftSide = document.getElementById('left-side');
+    leftSide.innerHTML = '';
+    const repoTable = createAndAppend('table', leftSide);
+    const repoTableBody = createAndAppend('tbody', repoTable);
+    addRepoInfoRow(
+      repoTableBody,
+      'Repository',
+      `<a href="${repository.html_url}" target="_blank">${repository.name}</a>`,
```

Comment:
You could set the `text` attribute to the empty string `''` and 'fix up' the last child if the table row after your call to `addRepoInfoRow`:

```js
const row = addRepoInfoRow(repoTableBody, 'Repository', '');
createAndAppend('a', row.lastChild, {
  href: repository.html_url,
  target: '_blank',
  text: repository.name,
});
```

Of course, your `addRepoInfoRow` function should then return the row it created as its return value.

Reply:
That is very useful way for escaping from using `html` in `createAndAppend` function. Many thanks for the tip.